### PR TITLE
feat (repository): Issue 120 : Upgrade the Issue Template to use GitHub Issue forms

### LINKED ISSUE #120: Upgrade the Issue Template to use GitHub Issue forms 📜
## Introduction
GitHub has recently rolled out a public beta for their issue forms feature. This would allow us to create interactive issue templates and validate them.

OpenEBS currently uses the older issue template format. The task is to create GitHub issue forms for this repository. 

## Tasks summary:
 - [ ] Fork & clone this repository
 - [ ] Prepare bug report issue form in .github/ISSUE_TEMPLATE/bug.yaml
 - [ ] Prepare documentation issue form in .github/ISSUE_TEMPLATE/documentation.yaml
 - [ ] Prepare feature request issue form in .github/ISSUE_TEMPLATE/feature.yaml
 - [ ] Push changes to master and test issue forms on your fork
 - [ ] Submit pull request

### PR DESCRIPTION
**Pull request by Rutam21**
# Description
This PR updates the Issue templates for `Feature`, `Bug`, and `Documentation` requests to use the latest Github Issue forms.

# Methodology
Respective `yaml` files have been added that would be used to provide the newest template for any issue requests.

**Checklist:**
- [x] Fixes #120 
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>

### CODE REVIEW

**Comment by Rutam21** [comment]
@kmova Please review and suggest changes, if any. Thanks.

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      required: true
+    attributes:
+      label: "🧱 Environment"
+      description: "Is your environment customized in any way?"
+      placeholder: "- OpenEBS version (use `kubectl get po -n openebs --show-labels`):
+                    - Kubernetes version (use `kubectl version`):
+                    - Cloud provider or hardware configuration:
+                    - OS (e.g: `cat /etc/os-release`):
+                    - kernel (e.g: `uname -a`):
+                    - others:"
```
Comment (vharsh) [inline .github/ISSUE_TEMPLATE/bug.yaml:90]: The LocalPV-provisioner is not a regular web-app per say, so things about the **browsers** & **operating systems** don't make sense, but you've added a very good point which can be beneficial for UX(and will likely also play well with bots which create issues **hopefully**) and we might use this for specifying K8s versions like `1.22` or K8s-version ranges.

File: .github/ISSUE_TEMPLATE/feature.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    id: environment
+    validations:
+      required: true
+    attributes:
+      label: "🧱 Environment"
+      description: "Is your environment customized in any way?"
+      placeholder: "- OpenEBS version (use `kubectl get po -n openebs --show-labels`):
+                    - Kubernetes version (use `kubectl version`):
+                    - Cloud provider or hardware configuration:
+                    - OS (e.g: `cat /etc/os-release`):
```
Comment (vharsh) [inline .github/ISSUE_TEMPLATE/feature.yaml:52]: this might go in a multi-line text-box type of input(if this is supported).

**Review by vharsh — COMMENTED** [review]
Thanks for sending out a PR, can you point us to the GitHub docs with various types of input fields? Also, I hope this way of structured issue creation is optional and the user creating the issue has the final say in leaving out some options.

Scenario: We might have made a dropdown for K8s versions, but issue may not be relevant for the version number in the drop-down, in this case the user should be able to specify an element which isn't in the dropdowns.

File: vendor/github.com/onsi/gomega/Dockerfile [hunk 3]
```diff
@@ -1 +1 @@
-FROM golang:1.15
+FROM golang:1.16
```
Comment (vharsh) [inline vendor/github.com/onsi/gomega/Dockerfile:1]: One shouldn't update files from the vendor directory, this vulnerability doesn't apply to us, but we'll make it a point to check the use of `golang:1.15` containers in the OpenEBS organization.

NOTE: We should avoid check-ing in the `vendor` as we have moved to Go Mod.

**Comment by niladrih** [comment]
Closing this due to inactivity.
Addtionally, this feature is not required. Thanks for the contribution @Rutam21 .

**Pull request closed without merging**